staging/src/k8s.io/client-go/tools/pager/pager.go

```diff
@@ -203,6 +203,11 @@ func (p *ListPager) eachListChunkBuffered(ctx context.Context, options metav1.Li
 	}()
 
 	for o := range chunkC {
+		select {
```

Comment:
I think it would be clearer to put the "already cancelled" case in the eachListChunk iterator above:

```diff
diff --git a/staging/src/k8s.io/client-go/tools/pager/pager.go b/staging/src/k8s.io/client-go/tools/pager/pager.go
index 9ba988f6856..7b8f5aaa032 100644
--- a/staging/src/k8s.io/client-go/tools/pager/pager.go
+++ b/staging/src/k8s.io/client-go/tools/pager/pager.go
@@ -194,20 +194,21 @@ func (p *ListPager) eachListChunkBuffered(ctx context.Context, options metav1.Li
                }()
                err = p.eachListChunk(ctx, options, func(chunk runtime.Object) error {
                        select {
-                       case chunkC <- chunk: // buffer the chunk, this can block
                        case <-ctx.Done():
+                               // context was already done, don't even send to the channel
                                return ctx.Err()
+                       default:
+                               select {
+                               case chunkC <- chunk: // buffer the chunk, this can block
+                               case <-ctx.Done():
+                                       return ctx.Err()
+                               }
                        }
                        return nil
                })
        }()
```